overlord/snapstate/storehelpers.go

```diff
@@ -800,28 +800,37 @@ func refreshCandidates(ctx context.Context, st *state.State, names []string, rev
 	return updates, stateByInstanceName, ignoreValidationByInstanceName, nil
 }
 
-// SnapHolds returns a map of held snaps to lists of holding snaps (including
-// "system" for user holds).
-func SnapHolds(st *state.State, snaps []string) (map[string][]string, error) {
-	var allSnapsHold string
+func effectiveRefreshHold(st *state.State) (time.Time, error) {
```

Comment:
thanks for introducing this but autorefresh.go is probably a better placement for this, if you introduce it just after EffectiveRefreshHold